app/services/manifest_submitter.rb

```diff
@@ -0,0 +1,26 @@
+class ManifestSubmitter
+  include Pundit
+  attr_reader :args
+
+  def initialize(args)
+    @args = args
+  end
+
+  def perform
+    output_stream = StreamLogger.new(Rails.logger)
+
+    manifest = Manifest.find_by_uuid_or_tracking_number!(args[:manifest_id])
+    authorize manifest, :can_submit?
```

Comment:
Fat controllers (the Sir Topham Hatt pattern) are not good, agreed.

I think it's important that we consistently apply authorization in the same place throughout the app. So far that has been in the controller layer, since that can reject some requests more quickly with appropriate HTTP-specific responses, rather than letting errors bubble up from deeper in the model layer.

The pattern with the services so far has been to let them assume no business rules (authorization) at all, but just to translate between the app and the CDX API. Middleware.

Param mangling, authn/authz checks: those are done in the controller layer elsewhere.